doc/userguide/devguide/extending/app-layer/overview.rst

```diff
@@ -0,0 +1,102 @@
+**************************
+Application Layer Overview
+**************************
+
+.. contents:: Table of Contents
+
+This section aims to give an overview of what is needed to add
+an application-layer protocol to Suricata.
+
+After a generic first step of collecting data about this application-layer protocol,
+especially pcaps for testing, we can dive into the Suricata specifics.
+
+An application-layer protocol has three logic components in Suricata :
+- parser
+- logger
+- detection engine
```

Comment:
`detection engine` reads a bit odd here since Detection Engine itself is a big component. wdyt?

Reply:
Detecting Keywords?